sdk/include/opentelemetry/sdk/metrics/state/attributes_hashmap.h

```diff
@@ -197,8 +209,13 @@ class AttributesHashMap
    */
   size_t Size() { return hash_map_.size(); }
 
+#ifdef UNIT_TESTING
+  size_t BucketCount() { return hash_map_.bucket_count(); }
+  size_t BucketSize(size_t n) { return hash_map_.bucket_size(n); }
+#endif
+
 private:
-  std::unordered_map<size_t, std::pair<MetricAttributes, std::unique_ptr<Aggregation>>> hash_map_;
+  std::unordered_map<MetricAttributes, std::unique_ptr<Aggregation>, CustomHash> hash_map_;
```

Comment:
can we store MetricAttributes as a pointer? it's going to be a lot copying on rehashing...

Reply:
Do you mean make the key as `std::shared_ptr<MetricAttributes>`, it may work with a more complicated custom hash. But I think `MetricsAttributes` should be created and moved into the `std::unordered_map`, so copying and rehashing should be in rare.

I remember the the exporting process does make a copy of `MetricsAttributes`, but there is an optimization for the common scenario. @lalitb any thoughts on this?

Reply:
I believe we are already avoiding deep copy with std::move while inserting the MetricsAttributes, also during rehashing, the hashmp will move(not copy) the MetricAttributes objects to their new locations. In general, I do see options for optimization at other places, but we can visit it separate to the PR.